docs/src/user_docs/assembly/io_operations.md

```diff
@@ -42,25 +42,21 @@ In both case the values must still encode valid field elements.
-| pushw.mem <br> pushw.mem.*a* <br> - *(6 cycles)* | [a, ... ] | [A, ... ] | $A \leftarrow mem[a]$ <br> Reads a word from memory at address $a$ and pushes it onto the stack. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
-| loadw.mem <br> - *(1 cycle)*  <br> loadw.mem.*a* <br> - *(2 cycles)*  | [a, 0, 0, 0, 0, ...] | [A, ... ] | $A \leftarrow mem[a]$ <br> Reads a word from memory at address $a$ and overwrites top four stack elements with it. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
-| pop.mem <br> - *(2 cycles)*  <br> pop.mem.*a*  <br> - *(3 cycles)*   | [a, v, ...] | [ ... ] | $[v, 0, 0, 0] \rightarrow mem[a]$ <br> Pops an element off the stack and stores it as the first element of the word in memory at address $a$. All other elements of the word are set to $0$. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
-| popw.mem <br> - *(5 cycles)*  <br> popw.mem.*a* <br> - *(6 cycles)*  | [a, A, ...] | [ ... ] | $A \rightarrow mem[a]$ <br> Pops the top four elements off the stack and stores them in memory at address $a$. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
-| storew.mem <br> - *(1 cycle)*  <br> storew.mem.*a* <br> - *(2 cycles)*  | [a, A, ...] | [A, ... ] | $A \rightarrow mem[a]$ <br> Stores the top four elements of the stack in memory at address $a$. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
+| <div style="width:100px">Instruction</div> | <div style="width:100px">Stack_input</div> | <div style="width:100px">Stack_output</div> |     Notes     |
+|                   :----:                   |                   :----:                   |                    :----:                   | ------------- |
+| mem_load <br> - *(1 cycle)*  <br> mem_load.*a* <br> - *(2 cycles)*   | [a, ... ] | [v, ... ] | $a \leftarrow mem[a][0]$ <br> Reads a word (4 elements) from memory at address *a*, and pushes the first element of the word onto the stack. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
+| mem_loadw <br> - *(1 cycle)*  <br> mem_loadw.*a* <br> - *(2 cycles)*  | [a, 0, 0, 0, 0, ... ] | [A, ... ] | $A \leftarrow mem[a]$ <br> Reads a word from memory at address $a$ and overwrites top four stack elements with it. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
+| mem_store <br> - *(1 cycle)*  <br> mem_store.*a*  <br> - *(2 cycles)*   | [a, v, ... ] | [v, ... ] | $[v, 0, 0, 0] \rightarrow mem[a]$ <br> Stores the top element of the stack as the first element of the word in memory at address $a$. All other elements of the word are set to $0$. If $a$ is provided via the stack, it is removed from the stack first. <br> Fails if $a \ge 2^{32}$ |
```

Comment:
This is no longe accurate: we don't set other elements to zeros, but rather retain their existing values. So, basically, we just overwrite the first element.